source/common/network/apple_dns_impl.cc

```diff
@@ -90,6 +90,10 @@ void AppleDnsResolverImpl::initializeMainSdRef() {
   // However, using a shared connection brings some complexities detailed in the inline comments
   // for kDNSServiceFlagsShareConnection in dns_sd.h, and copied (and edited) in this implementation
   // where relevant.
+  //
+  // When error occurs while the main_sd_ref_ is initialized, the initialize_failure_timer_ will be
+  // enabled to retry initialization. Retries an also be triggered via query submission, @see
```

Comment:
```suggestion
  // enabled to retry initialization. Retries can also be triggered via query submission, @see
```

Reply:
Can you also add a comment below about why/when `dnsServiceRefSockFD` fails? Seems bizarre to me that would fail.

Reply:
Same, but unfortunately the header docs don't give any detail. I added a comment.